dubbo-remoting/dubbo-remoting-http/src/main/java/org/apache/dubbo/remoting/http/restclient/OKHttpRestClient.java

```diff
@@ -136,15 +150,35 @@ public void close(int timeout) {}
 
     @Override
     public boolean isClosed() {
-        return okHttpClient.retryOnConnectionFailure();
+        return false;
     }
 
-    public OkHttpClient createHttpClient(HttpClientConfig httpClientConfig) {
+    public OkHttpClient createHttpClient(HttpClientConfig httpClientConfig, URL url) {
+        // reuse  okHttpClient
+        if (this.okHttpClient != null) {
+            return okHttpClient;
+        }
+
+        Dispatcher dispatcher = new Dispatcher();
+        if (url != null) {
+            // use dubbo client thread pool to replace okhttp self thread pool
+            dispatcher = new Dispatcher(initRestClientExecutor(url));
+        }
+
         OkHttpClient client = new OkHttpClient.Builder()
+                .dispatcher(dispatcher)
                 .readTimeout(httpClientConfig.getReadTimeout(), TimeUnit.SECONDS)
                 .writeTimeout(httpClientConfig.getWriteTimeout(), TimeUnit.SECONDS)
                 .connectTimeout(httpClientConfig.getConnectTimeout(), TimeUnit.SECONDS)
                 .build();
         return client;
     }
+
+    private ExecutorService initRestClientExecutor(URL url) {
+        ExecutorRepository executorRepository = ExecutorRepository.getInstance(url.getOrDefaultApplicationModel());
+
+        url = url.addParameter(THREAD_NAME_KEY, CLIENT_THREAD_POOL_NAME)
+                .addParameterIfAbsent(THREADPOOL_KEY, DEFAULT_CLIENT_THREADPOOL);
+        return executorRepository.createExecutorIfAbsent(url);
+    }
```

Comment:
Should support isolated executors here.

[1] `org.apache.dubbo.rpc.executor.ExecutorSupport`
[2] `org.apache.dubbo.remoting.transport.dispatcher.WrappedChannelHandler#getSharedExecutorService(java.lang.Object)`
[3] `org.apache.dubbo.rpc.protocol.tri.h12.http2.GenericHttp2ServerTransportListener#initializeExecutor`